core/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -167,7 +183,12 @@ public String analyzer() {
         return this.analyzer;
     }
 
-    /** Sets a slop factor for phrase queries */
+    /**
+     * Sets a slop factor for phrase queries
+     *
+     * @deprecated for phrase queries use {@link MatchPhraseQueryBuilder}
+     */
+    @Deprecated
```

Comment:
We still need `MatchQuery.Type` as we are still using `MatchQuery` underneath and need to tell it which mode to use.